src/spatial_network.jl

```diff
@@ -0,0 +1,210 @@
+
+
+# This function takes a graph and a set of indices
+# and returns an iterator over assignments
+# i -> e[1], j -> e[2], k -> e[3], ...
+# for all n-edges of the graph, where n is the
+# number of indices. The assignments should be
+# of type Dict{Symbol,Symbol}.
+#
+# For an ordinary graph, the 1-edges are the vertices
+# and the 2-edges are the edges in the usual sense.
+#
+# Important: as the order of the indices is not
+# defined, this function should assign elements
+# to all permutations of the edges, ie. for
+# i => 4, j => 5 it should also return i => 5, j => 4
+function graph_iterator end
+
+struct BasicGraph{T}
```

Comment:
We should really just build this on LightGraphs.jl instead of building out a whole new graph format.